[BUG] test failure: test_unicode.UnicodeTest.test_raiseMemError on 32bit in Debian CI

### Describe the bug

`test_unicode.UnicodeTest.test_raiseMemError` is failing on i386, armhf and armel (the 32 bit ARM ports) in Debian's CI.

Logs: https://ci.debian.net/packages/c/cython/

The cause isn't obvious so we'll skip the test, but I'm filing this so that someone can investigate more deeply.

### Code to reproduce the behaviour:

_No response_

### Expected behaviour

_No response_

### OS

Linux (Debian)

### Python version

3.12.1

### Cython version

3.0.6

### Additional context

```
======================================================================
FAIL: test_raiseMemError (test_unicode.UnicodeTest.test_raiseMemError)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/run/test_unicode.pyx", line 2355, in test_unicode.UnicodeTest.test_raiseMemError (test_unicode.c:144223)
    self.assertRaises(MemoryError, alloc)
AssertionError: MemoryError not raised by lambda

======================================================================
FAIL: test_raiseMemError (test_unicode.UnicodeTest.test_raiseMemError)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/run/test_unicode.pyx", line 2355, in test_unicode.UnicodeTest.test_raiseMemError (test_unicode.cpp:144235)
    self.assertRaises(MemoryError, alloc)
AssertionError: MemoryError not raised by lambda
```

--- a/tests/run/test_unicode.pyx
+++ b/tests/run/test_unicode.pyx
@@ -2326,6 +2326,10 @@ class UnicodeTest(CommonTest,
         s = 'abc'
         self.assertIs(s.expandtabs(), s)
 
+    """
+    # Not useful for Cython: struct sizes change between versions
+    # so it's hard to keep reliably up-to-date, and it's largely checking
+    # a CPython implementation detail
     def test_raiseMemError(self):
         if struct.calcsize('P') == 8:
             # 64 bits pointers
@@ -2354,6 +2358,7 @@ class UnicodeTest(CommonTest,
             alloc = lambda: char * maxlen
             self.assertRaises(MemoryError, alloc)
             self.assertRaises(MemoryError, alloc)
+        """
 
     def test_format_subclass(self):
         class S(str):

